test/test_functional.py

```diff
@@ -247,6 +251,59 @@ def test_create_fb(self):
         self._test_create_fb(n_mels=56, fmin=1900.0, fmax=900.0)
         self._test_create_fb(n_mels=10, fmin=1900.0, fmax=900.0)
 
+    def test_gain(self):
+        waveform, sample_rate = torchaudio.load(self.test_filepath)
+        waveform_gain = F.gain(waveform, 5)
+        self.assertTrue(waveform_gain.abs().max().item(), 1.)
+
+        test_filepath_sox_gain = os.path.join(self.test_dirpath, "assets", "sinewave_soxgain5.wav")
+        sox_gain_waveform, sox_sr = torchaudio.load(test_filepath_sox_gain)
+
+        self.assertTrue(torch.allclose(waveform_gain, sox_gain_waveform))
+
+    def test_scale_to_interval(self):
+        waveform, sample_rate = torchaudio.load(self.test_filepath)
+        scaled = 5.5  # [-5.5, 5.5]
+        waveform_scaled = F.scale_to_interval(waveform, scaled)
+
+        self.assertTrue(torch.max(waveform_scaled) <= scaled)
+        self.assertTrue(torch.min(waveform_scaled) >= -scaled)
+
+    def test_dither(self):
+        waveform, sample_rate = torchaudio.load(self.test_filepath)
+        waveform_dithered = F.dither(waveform)
+        waveform_dithered_noiseshaped = F.dither(waveform, noise_shaping=True)
+
+        test_filepath_sox_dither = os.path.join(self.test_dirpath, "assets", "sinewave_soxdither.wav")
+        sox_dither_waveform, sox_sr = torchaudio.load(test_filepath_sox_dither)
+
+        self.assertTrue(torch.allclose(waveform_dithered, sox_dither_waveform, rtol=1e-03, atol=1e-03))
+
+        test_filepath_sox_dither_ns = os.path.join(self.test_dirpath, "assets", "sinewave_soxdither_noiseshaping.wav")
+        sox_dither_waveform_ns, sox_sr = torchaudio.load(test_filepath_sox_dither_ns)
+
+        self.assertTrue(torch.allclose(waveform_dithered_noiseshaped, sox_dither_waveform_ns, rtol=1e-03, atol=1e-03))
+
+    def test_vctk(self):
+        test_filepath_vctk = os.path.join(self.test_dirpath, "assets/VCTK-Corpus/wav48/p224/", "p224_002.wav")
+        wf_vctk, sr_vctk = torchaudio.load(test_filepath_vctk)
+
+        # rate
+        sample = T.Resample(sr_vctk, 16000, resampling_method='sinc_interpolation')
+        wf_vctk = sample(wf_vctk)
+        # dither
+        wf_vctk = F.dither(wf_vctk, noise_shaping=True)
+
+        test_filepath_vctk_sox = os.path.join(self.test_dirpath, "assets", "p224_002_sox.wav")
+        # E.append_effect_to_chain("gain", ["-h"])
+        # E.append_effect_to_chain("channels", [1])
+        # E.append_effect_to_chain("rate", [16000])
+        # E.append_effect_to_chain("gain", ["-rh"])
+        # E.append_effect_to_chain("dither", ["-s"])
+        waveform_vctk_sox, sr_vctk_sox = torchaudio.load(test_filepath_vctk_sox)
```

Comment:
Where are the sox filters applied?

Reply:
I did this in the terminal, so the .wav file is the output

Reply:
I'd make this a runtime operation during test time. See the other tests. sox itself might be faulty or change (across platforms) and we want to track that as well.

Reply:
Cool, looks like that's what @CamiWilliams did :)